### ISSUE TITLE
[post frontend-backend split] Advice with sealed phase in the frontend

### ISSUE THREAD

**Issue by ed255** [privacy-scaling-explorations/halo2#295]
Currently the `Advice` type used in the frontend comes from the middleware, which exposes the phase as a `u8`.  In order to hide this from the user we need to have copy of the `Advice` type internal to the frontend with a sealed phase.

Probably this means also brining duplicates for `Fixed`, `Instance`, `ColumnType` and `Any` into the frontend.  After this, the corresponding types in the middleware can be greatly simplified.

Blocked by https://github.com/privacy-scaling-explorations/halo2/pull/290

**Comment by duguorong009** [comment]
I believe this issue has already been addressed in #300 .
Hence, I propose that we close this issue as it is resolved or no longer needed.

What do you think? @ed255

**Comment by ed255** [comment]
> I believe this issue has already been addressed in #300 . Hence, I propose that we close this issue as it is resolved or no longer needed.
> 
> What do you think? @ed255

You're totally right!  Thanks for checking this, I'll just close the issue :)
